src/index.ts

```diff
@@ -350,11 +374,16 @@ class MagicNixCacheAction extends DetSysAction {
     }
 
     actionsCore.debug(`killing`);
+
     try {
       process.kill(pid, "SIGTERM");
-    } catch (e) {
+    } catch (e: unknown) {
       if (typeof e === "object" && e && "code" in e && e.code !== "ESRCH") {
-        throw e;
+        if (this.strictMode) {
+          throw e;
```

Comment:
There seems to be a lot of this "if strict then ... else ..." -- would it make sense to abstract that into a function?

Reply:
The usages are subtly different enough that I think it's best to keep it like it is